raiden/tests/integration/api/rest/test_payments.py

```diff
@@ -148,6 +148,107 @@ def test_api_payments(
+    token_addresses,
+    deposit,
+) -> None:
+    _, app1 = raiden_network
+    amount = 100
+    identifier = 42
+    token_address = token_addresses[0]
+    target_address = app1.address
+
+    our_address = api_server_test_instance.rest_api.raiden_api.address
+    our_metadata = api_server_test_instance.rest_api.raiden_api.raiden.transport.address_metadata
+
+    payment = {
+        "initiator_address": to_checksum_address(our_address),
+        "target_address": to_checksum_address(target_address),
+        "token_address": to_checksum_address(token_address),
+        "amount": str(amount),
+        "identifier": str(identifier),
+    }
+
+    # Test a normal payment
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={
+            "amount": str(amount),
+            "identifier": str(identifier),
+            "paths": [
+                {
+                    "route": [to_checksum_address(our_address), to_checksum_address(app1.address)],
+                    "address_metadata": {
+                        to_checksum_address(our_address): our_metadata,
+                        to_checksum_address(app1.address): app1.transport.address_metadata,
+                    },
+                }
+            ],
+        },
+    )
+
+    with watch_for_unlock_failures(*raiden_network):
+        response = request.send().response
+    assert_proper_response(response)
+    json_response = get_json_response(response)
+    assert_payment_secret_and_hash(json_response, payment)
+
+    # Test a payment without providing an identifier
+    payment["amount"] = "1"
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={"amount": "1"},
+    )
+    with watch_for_unlock_failures(*raiden_network):
+        response = request.send().response
+    assert_proper_response(response)
+    json_response = get_json_response(response)
+    assert_payment_secret_and_hash(json_response, payment)
+
+    # Test that trying out a payment with an amount higher than what is available
+    # returns an error
+    payment["amount"] = str(deposit)
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={"amount": str(deposit)},
+    )
+    response = request.send().response
+    assert_proper_response(response, status_code=HTTPStatus.CONFLICT)
```

Comment:
This part of the test would make sense for `test_api_payments_without_pfs` if we would actually provide the `paths` argument and then check that the path validation leads to the correct conflict.

Reply:
✅